### DISCUSSION TITLE
Ability to group header title on the fly and collapse and expand them too

### DISCUSSION THREAD

**Discussion by ptatoChi** [ghiscoding/Angular-Slickgrid#1283] [Ideas]
Hi. I like the Header Title Grouping feature. 
![image](https://github.com/ghiscoding/Angular-Slickgrid/assets/11557985/a19c3c71-2362-4b0c-b5d2-d13b7c8c6d0d)

I was hoping if i could dynamically group columns titles when the table is already generated. and able to collapse and expand those groups

**Comment by ghiscoding** [comment]
You should try something before asking because it doesn't look like you tried anything since you didn't provide the code you tried. Anyway the events that triggers a re-rendering of the column grouping is [here](https://github.com/ghiscoding/slickgrid-universal/blob/78abafff21e3b5b3df1ac5413969487eeb788fcb/packages/common/src/services/groupingAndColspan.service.ts#L64-L69). I guess you could also trigger the same method yourself to re-render by calling `renderPreHeaderRowGroupingTitles()` from the GroupingAndColspanService service instance

You also should consider asking these type of questions on Stack Overflow, you should always provide code of what you tried

**Reply by ptatoChi** [reply]
Thanks for the reply. I was trying to use the title grouping, but it doesn't work when draggable grouping is enabled. is there a reason why?

**Reply by ghiscoding** [reply]
yes because they both use the pre-header section and there is only 1 such section, so it cannot work

**Reply by ptatoChi** [reply]
Sorry for this. Its just a clarification. So youre saying that I can implement the dynamic creating a header title group by selecting multiple headers, rightclick, create header group, then enter title for that group?

It would be nice if those header title groups can be collapsible (like from the example picture above, clicking `Period` group header can hide and show the headers belonging to it?

**Reply by ghiscoding** [reply]
What you are trying to do will **not work** because there is only 1 pre-header section in SlickGrid and only 1 feature can use it but you are trying to use 2 features in the same div section, it will never work. I didn't create SlickGrid and I don't know why they decided to use the same section but it is what it is and you cannot use these 2 features at the same time for that reason

![image](https://github.com/ghiscoding/Angular-Slickgrid/assets/643976/1fd38a24-ef49-45f8-bb5a-cd9d4382f14e)

![image](https://github.com/ghiscoding/Angular-Slickgrid/assets/643976/255279f1-0ef7-4abe-b05e-e1d6bf9c0db5)

You cannot use them at the same time, one will overwrite the other
